extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java

```diff
@@ -572,55 +574,100 @@ static Match validateNestedExpressions(TemplateAnalysis templateAnalysis, ClassI
         }
 
         while (iterator.hasNext()) {
-            // Now iterate over all parts of the expression and check each part against the current "match class"
+            // Now iterate over all parts of the expression and check each part against the current match type
             Info info = iterator.next();
             if (!match.isEmpty()) {
-                // By default, we only consider properties
-                Set<String> membersUsed = implicitClassToMembersUsed.get(match.clazz().name());
-                if (membersUsed == null) {
-                    membersUsed = new HashSet<>();
-                    implicitClassToMembersUsed.put(match.clazz().name(), membersUsed);
+
+                // Arrays are handled specifically
+                // We use the built-in resolver at runtime because the extension methods cannot be used to cover all combinations of dimensions and component types 
+                if (match.isArray()) {
+                    if (info.isProperty()) {
+                        String name = info.asProperty().name;
+                        if (name.equals("length")) {
+                            // myArray.length
+                            match.setValues(null, PrimitiveType.INT);
+                            continue;
+                        } else {
+                            // myArray[0], myArray.1
+                            try {
+                                Integer.parseInt(name);
+                                match.setValues(null, match.type().asArrayType().component());
+                                continue;
+                            } catch (NumberFormatException e) {
+                                // not an integer index
```

Comment:
Shouldn't this error break the build since the template is malformed?

Reply:
Not really. In theory, you could have an extension method that adds more functionality to arrays, for example: https://github.com/quarkusio/quarkus/pull/14793/files#diff-77caa2024cb8bf43e6ff316cb8f6c9af1a6963fd0e686165cc76acdd2aa3dbe8R154-R156